.travis.yml

```diff
@@ -1,33 +1,61 @@
 language: php
 
-sudo: false
-
 cache:
   directories:
     - $HOME/.composer/cache/files
 
+branches:
+    only:
+        - master
+
 php:
-  - 5.5
-  - 5.6
-  - 7.0
-  - hhvm
+  - '5.5'
+  - '5.6'
+  - '7.0'
+  - '7.1'
+  - nightly
 
 matrix:
   fast_finish: true
   include:
-    - php: 5.6
-      env: DEPS=dev
-    - php: 5.5
+    - php: '5.5'
       env: COMPOSER_FLAGS="--prefer-lowest"
+
+    - php: '7.0'
+      env: 'SYMFONY_VERSION=~2.8.0'
     - php: 7.0
-      env: SYMFONY_VERSION=2.7.*
+      env: 'SYMFONY_VERSION=~3.0.0'
     - php: 7.0
-      env: SYMFONY_VERSION=2.8.*
+      env: 'SYMFONY_VERSION=~3.1.0'
+    - php: 7.1
+      env: 'SYMFONY_VERSION=~3.2.0@dev'
+
+    - php: 7.0
+      env: 'ELOQUENT_VERSION=~5.0.0'
+    - php: 7.0
+      env: 'ELOQUENT_VERSION=~5.1.0'
+    - php: 7.0
+      env: 'ELOQUENT_VERSION=~5.2.0'
+    - php: 7.0
+      env: 'ELOQUENT_VERSION=~5.3.0@dev'
+
+  allow_failures:
+    - php: nightly
+    - env: 'ELOQUENT_VERSION=~5.3.0@dev'
 
 before_install:
+  - set -eo pipefail
   - phpenv config-rm xdebug.ini || true
-  - composer self-update
-  - if [ "$DEPS" = "dev" ]; then perl -pi -e 's/^}$/,"minimum-stability":"dev"}/' composer.json; fi
-  - if [ "$SYMFONY_VERSION" != "" ]; then composer require symfony/symfony:${SYMFONY_VERSION} --no-update; fi
+  - if [ -n "$SYMFONY_VERSION" ]; then
+        composer require --no-update "symfony/symfony:${SYMFONY_VERSION}";
+    fi;
+  - if [ -n "$ELOQUENT_VERSION" ]; then
+        composer require --no-update "illuminate/database:${ELOQUENT_VERSION}";
+    fi;
 
 install: composer update --prefer-dist $COMPOSER_FLAGS
+
+script: vendor/bin/phpunit
+
+notifications:
+  email: false
```

Comment:
err dunno, tbh I don't get any emails it's for you so if you prefer to have them I can just remove that :)